### ISSUE TITLE
Upstreaming of GNU tools patches.

### ISSUE THREAD

**Issue by joshdekock** [pspdev/psptoolchain#73]
I'd like to attempt to upstream the patches to preserve a bit of digital history, and to ensure that PSP support in the GNU toolchain is always up-to-date (I will take up maintainership of the PSP specifics in the GNU toolchain).
I see there were some efforts to upstream the psptoolchain patches, but they hit a roadblock because of no one trying to get a assignment to copyright FSF or public domain.
In order for the patches to be upstreamed there are some legal prerequisites. Please don't hesitate to ask if you have any queries.

Would you be willing to assign the copyright to the Free Software Foundation or sign a copyright disclaimer to put this change in the public domain, so that it could be integrated into the patches' respective package? (This means if you contributed to the GCC patch then you release your contributions into the public domain or assign copyright to the FSF, same for the binutils patch etc).

Below is a list of the contributors I have found who would need to agree to the terms in order to make this a reality.

GitHub Contributors:
- [ ] @Bracket-
- [ ] @elsonLee
- [ ] @SopaXorzTaker
- [ ] @WinterMute
- [ ] @himanshugoel2797
- [ ] @davidgfnet
- [ ] @dogo
- [ ] @FTPiano 
- [ ] @xantares
- [ ] @n00neimp0rtant
- [ ] @caffinatedmonkey
- [ ] @SamRH
- [ ] @artart78
- [ ] @uppfinnarn 
- [ ] @take-cheeze

Other git contributors: (hopefully no duplicates)
- [ ] Arthur Blot
- [ ] Clément Guérin
- [ ] Dan Peori
- [ ] Dave Murphy
- [ ] David Guillen Fandos
- [ ] Diogo Autilio
- [ ] Johannes Ekberg
- [ ] Martin Charles
- [ ] Scott Marnik
- [ ] Victor Xie
- [ ] elson
- [ ] nem

Original ps2dev contributors:
- [ ] Mr Brown
- [ ] Pixel
- [ ] Lukasz
- [ ] ooPo

**Comment by SopaXorzTaker** [comment]
My changes are pretty minor and I am not against licensing them in any way. I explicitly allow my changes to be released into public domain.

**Comment by himanshugoel2797** [comment]
I agree. My changes are also extremely minor and am not against releasing my changes to the public domain.

I release my contributions to the psptoolchain patches change into the public domain.

**Comment by artart78** [comment]
I release my contributions to the psptoolchain patches change into the public domain.

**Comment by dogo** [comment]
I release my contributions to the psptoolchain patches change into the public domain.

**Comment by xantares** [comment]
I release my contributions to the psptoolchain patches change into the public domain.

**Comment by FTPiano** [comment]
I release my contributions to the psptoolchain patches change into the public domain.

**Comment by caffinatedmonkey** [comment]
My changes are minor also so:

I assign copyright to the Free Software Foundation of all my contributions to the psptoolchain patches.

**Comment by take-cheeze** [comment]
I release my contributions to the psptoolchain patches change into the public domain.

**Comment by davidgfnet** [comment]
Just noticed I'm a duplicate (davidgfnet & David Guillen Fandos), sorry for
that!

2016-10-24 9:48 GMT+02:00 David Guillen <david@davidgf.net>:

**Comment by davidgfnet** [comment]
Hey, thanks for your efforts man. I really appreciate you doing this, it's
important to preserve history in the world we live today.

I release any of my work there into the public domain.


2016-10-23 11:27 GMT+02:00 Takeshi Watanabe <notifications@github.com>:

**Comment by libcg** [comment]
I release my contributions to the psptoolchain patches change into the public domain.

**Comment by WinterMute** [comment]
@WinteMute & Dave Murphy are both me & I have an FSF copyright assignment for binutils, gcc & newlib on file. For any other packages I release my contributions to the psptoolchain patches into the public domain.

**Comment by John-K** [comment]
If any of the VFPU stuff I did is still in there, or other code:
I release my contributions to the psptoolchain patches change into the public domain.

**Comment by Bracket-** [comment]
I release my contributions to the psptoolchain patches change into the public domain.

**Comment by uppfinnarn** [comment]
I release my contributions to the psptoolchain patches change into the public domain.

I'm #1 on the "other git contributors" list, but I'd prefer if you'd use my username instead.

**Comment by ooPo** [comment]
I release my contributions to the psptoolchain patches change into the public domain.

Good luck, it'll be awesome if you can pull this off. :)

**Comment by lukaszdk** [comment]
I release my contributions to the psptoolchain patches change into the public domain.Den 16. dec. 2016 19.40 skrev Josh de Kock <notifications@github.com>:I'd like to attempt to upstream the patches to preserve a bit of digital history, and to ensure that PSP support in the GNU toolchain is always up-to-date (I will take up maintainership of the PSP specifics in the GNU toolchain).
I see there were some efforts to upstream the psptoolchain patches, but they hit a roadblock because of no one trying to get a assignment to copyright FSF or public domain.
In order for the patches to be upstreamed there are some legal prerequisites. Please don't hesitate to ask if you have any queries.
Would you be willing to assign the copyright to the Free Software Foundation or sign a copyright disclaimer to put this change in the public domain, so that it could be integrated into the patches' respective package? (This means if you contributed to the GCC patch then you release your contributions into the public domain or assign copyright to the FSF, same for the binutils patch etc).
Below is a list of the contributors I have found who would need to agree to the terms in order to make this a reality.
GitHub Contributors:
 @Bracket- @elsonLee @SopaXorzTaker @WinterMute @himanshugoel2797 @davidgfnet @dogo @FTPiano @xantares @n00neimp0rtant @caffinatedmonkey @SamRH @artart78 @uppfinnarn @take-cheeze @libcg @John-K @ooPo @igetgames @Lukaszdk
Other git contributors: (hopefully no duplicates)
 Scott Marnik elson nem
Original ps2dev contributors:
 Mr Brown Pixel Lukasz
Reply with either:
I assign copyright to the Free Software Foundation of all my contributions to the psptoolchain patches.

I release my contributions to the psptoolchain patches change into the public domain.

No, I don't want this. (With a reason, preferably.)

**Comment by igetgames** [comment]
I release my contributions to the psptoolchain patches change into the public domain.

I am also listed under 'Original ps2dev contributors' as 'Mr Brown' (I used my abbreviated full name, M. R. Brown in my copyright notices, and went by the handle 'mrbrown'), so this release also covers that pseudonym.

**Comment by SamRH** [comment]
I release my contributions to the psptoolchain patches change into the public domain.

**Comment by eintw1ck** [comment]
I have emailed the last two contributors with actual contact information, @n00neimp0rtant and @elsonLee. I will have a look at the git history to see how considerable the contribution by the remaining unknown persons. It might be ok to just call it public domain after a certain amount of the code is (larger projects owned by organisations such as Mozilla have done this).

**Comment by mirh** [comment]
So, is this going anywhere?

**Comment by davidgfnet** [comment]
Yeah same question. Anything we can do? I'd love to see this upstreamed, really!

**Comment by jdek** [comment]
We are still missing some contributors releasing their into the public domain/copyright assignment, but I looked into it further and it seems like we might really struggle to upstream it specifically. If you can help contact these last people that would be the final things which would need to be done. I have tried to in the past without success.

Regardless, I think the goal should switch here to just getting everything public domain and then slapping an `UNLICENSE` file in, preserving it as a bit of unencumbered digital history. For what it's worth, everyone in the list so far has already put their work in the public domain so we are just missing a couple. This, to me, would be a good enough compromise.

**Comment by davidgfnet** [comment]
Looking at some of the contribs I'd call some trivial, perhaps not even worth discussing. I mean adding 5 lines into a few thousand line soup can't be realistictly claimable. Particularly since most of the contribs are not really source (but scripts, yeah well it's source but you know what i mean).
About upstreaming: is it a licensing problem the one you see? I was afraid the problems might be related to other things (like projects rejecting these contributions).

**Comment by jdek** [comment]
The issue I found was more that it's 'few thousand line soup' worth of patches, and would need to be refactored rather heavily. I won't have time to do this anymore really: since I opened this issue, I finished school (twice) and started work :p

Licensing might still be an issue since some of the contributors didn't assign copyright to the FSF but rather just put their work into the public domain. The FSF doesn't particularly like this and if it's something they're not particularly keen on taking in anyway (from a maintainer's point of view) then they might just say 'we don't like the licensing situation'.

**Comment by nicolasnoble** [comment]
Pixel here. I release my contributions to the psptoolchain patches change into the public domain.

**Comment by wally4000** [comment]
It looks as if some of these non-responding contributions are directly related to the toolchain script itself rather than part of the upstreaming, is it possible that we can ignore these contributions as they're not directly related to patches for gcc / binutils?
